core/src/main/java/org/bitcoinj/wallet/Wallet.java

```diff
@@ -4648,6 +4675,24 @@ public void completeTx(SendRequest req) throws InsufficientMoneyException, BadWa
         }
     }
 
+    /**
+     * Removes candidates that are already included as inputs
+     * in the SendRequest. This is to avoid using the same
+     * input twice in a Transaction.
+     */
+    private static List<TransactionOutput> removeExistingInputs(List<TransactionOutput> candidates, SendRequest sendRequest) {
+        List<TransactionInput> txInputs = sendRequest.tx.getInputs();
+        return candidates.stream().filter(candidate -> {
+
+            boolean candidateNotExistingInput = !txInputs.stream().anyMatch(txInput -> {
+                TransactionOutPoint outpoint = txInput.getOutpoint();
+                return Objects.equals(candidate.getParentTransactionHash(), outpoint.hash()) &&
+                        candidate.getIndex() == outpoint.index();
+            });
+            return candidateNotExistingInput;
```

Comment:
I would recommend moving this filter lambda to a private helper method for readability.